modules/code-editor/src/views/full/LiteEditor/index.tsx

```diff
@@ -225,13 +280,38 @@ export default class MinimalEditor extends React.Component<Props> {
     monaco.languages.typescript.typescriptDefaults.addExtraLib(data, 'bp://types/custom_variables.d.ts')
   }
 
+  getHints = (scope: string) => {
+    if (!this.props.hints) {
+      return []
+    }
+
+    return this.props.hints
+      .filter(x => x.name.startsWith(scope))
+      .map(x => `/** ${x.source}. ${x.location} */\n${x.name.replace(`${scope}.`, '')}: string\n`)
```

Comment:
can you extract the function in the map with a nice naming so we get a clear understanding of what's happening here